pkg/envoy/cds/builder.go

```diff
@@ -28,9 +28,139 @@ import (
 	"github.com/openservicemesh/osm/pkg/trafficpolicy"
 )
 
+type clusterBuilder struct {
+	proxyIdentity identity.ServiceIdentity
+
+	outboundMeshTrafficPolicy *trafficpolicy.OutboundMeshTrafficPolicy
+
+	inboundMeshTrafficPolicy *trafficpolicy.InboundMeshTrafficPolicy
+
+	sidecarSpec configv1alpha2.SidecarSpec
```

Comment:
should this be a more generic set of configuration options?  Sidecar spec sounds like k8s object?

Reply:
that's a great question! I'd argue that's a bit out of scope, and the sidecarspec is actually defined by us, so if in a world where OSM ran outside of K8s, I think it would be reasonable for us to define our required API's, which could still include the SidecarSpec

But I concede it takes some squinting to get to that conclusion :)

Reply:
makes sense. The `Sidecar` terminology through me off a bit but looking at struct it seems like it has some generic configuration for the "proxy". It does have some things that are a bit overly specific to kubernetes but we do own it and could tweak it in the future.  Thanks!